chroma-manager/tests/integration/utils/chroma_log_collector.py

```diff
@@ -120,67 +142,97 @@ def fetch_log(self, server, source_log_path, destination_log_filename):
+        # Find the diagnostics filename from the iml-diagnostics output
+        cd_out = result.stdout.decode('utf8')
+        match = re.compile('/var/tmp/(sosreport-.*\.tar\..*)').search(cd_out)
         if not match:
-            return ["Did not find diagnostics filepath in chroma-diagnostics output:\nstderr:\n%s\nstdout:\n%s" %
-                    (cd_out, result.stdout.decode('utf8'))]
+            return [
+                "Did not find diagnostics filepath in iml-diagnostics output:\nstderr:\n%s\nstdout:\n%s"
+                % (cd_out, result.stdout.decode('utf8'))
+            ]
         diagnostics = match.group(1).strip()
 
         errors = []
 
-        # Copy and expand the diagnostics locally, so they will be able to be read in browser in Jenkins.
-        errors.append(self.fetch_log(server, "/var/log/%s" % diagnostics, ''))
-        errors.append(self.fetch_log(server, "chroma-diagnostics.log", '%s-chroma-diagnostics.log' % server))
-
-        if diagnostics.endswith('tar.lzma'):
-            if shell_run(['tar', '--lzma', '-xvf', "%s/%s" % (self.destination_path, diagnostics),
-                               '-C', self.destination_path]).rc:
-                errors.append("Error tar --lzma the chroma diagnostics file")
+        # Copy and expand the diagnostics locally, so they will be
+        # able to be read in browser in Jenkins.
+        errors.append(self.fetch_log(server, "/var/tmp/%s" % diagnostics, ''))
+
+        if diagnostics.endswith('tar.xz'):
+            if shell_run([
+                    'tar', '-xvJf',
+                    "%s/%s" % (self.destination_path, diagnostics), '-C',
+                    self.destination_path
+            ]).rc:
+                errors.append("Error tar --xvJf the iml diagnostics file")
         elif diagnostics.endswith('tar.gz'):
-            if shell_run(['tar', '-xvzf', "%s/%s" % (self.destination_path, diagnostics),
-                               '-C', self.destination_path]).rc:
-                errors.append("Error tar -xvzf the chroma diagnostics file")
+            if shell_run([
+                    'tar', '-xvzf',
+                    "%s/%s" % (self.destination_path, diagnostics), '-C',
+                    self.destination_path
+            ]).rc:
+                errors.append("Error tar -xvzf the iml diagnostics file")
         else:
-            errors = "Didn't recognize chroma-diagnostics file format"
-
-        if shell_run(['rm', '-f', "%s/%s" % (self.destination_path, diagnostics)]).rc:
-            errors.append("Unable to remove the diagnostics %s/%s" % (self.destination_path, diagnostics))
+            errors = "Didn't recognize iml-diagnostics file format"
+
+        diagnostics_dir = re.compile('(sosreport-.*)\.tar\..*').search(
+            diagnostics).group(1)
+
+        try:
+            os.chmod(
+                "%s/%s/sys/module/md_mod/parameters/new_array"
+                % (self.destination_path, diagnostics_dir),
+                0777
```

Comment:
So this is the only file that was choking the `chmod -R 777`?  What is it's permission flags before the `chmod()`?